clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestState.java

```diff
@@ -127,8 +127,16 @@ public void onSuccessfulAttempt(final long currentTimeMs) {
      * @param currentTimeMs Current time in milliseconds
      */
     public void onFailedAttempt(final long currentTimeMs) {
+        onFailedAttempt(currentTimeMs, false);
+    }
+
+    public void onFailedAttempt(final long currentTimeMs, final boolean skipBackoff) {
         this.lastReceivedMs = currentTimeMs;
-        this.backoffMs = exponentialBackoff.backoff(numAttempts);
+        if (skipBackoff) {
+            this.backoffMs = 0;
+        } else {
+            this.backoffMs = exponentialBackoff.backoff(numAttempts);
+        }
         this.numAttempts++;
```

Comment:
I did consider using the `onSuccessfulAttempt`, but did not like it too much because even though it does what we need, it did not seem conceptually right since it's not exactly a successful case when we get those errors, and that could maybe lead to trouble in the future if we start taking some more actions on the onSuccessful path I expect. I did not think about the`resetBackoff` one, and I actually like it, will go for it and include the `numAttempts` in the reset, makes sense.